### DISCUSSION TITLE
[v4] Lifespan

### DISCUSSION THREAD

**Discussion by davebot9000** [adonisjs/core#2416] [Help] [answered]
Sorry if this question has been asked/answered before (github discussions search features are kinda trash). Is there an official cut off to when Adonis v4 will stop getting updates (e.g. security, bug fixes etc)? Just trying to plot out my migration to v5. Really any clarification on when it is intended to be retired, no longer maintained etc would be great. I looked at the roadmap on trello and didn't see any info, also have tried to google for this info, with no luck. 

Thank you!

**Comment by thetutlage** [comment, marked as answer]
Hello @dpr989 

We are already in the process of releasing v5. It can be next week or maybe the week after that. Just before the v5 release, we will update all the v4 packages pushing bug fixes and updating security critical underlying dependencies. Post that, v4 will not receive any updates, however the docs will be live.

If you are starting a new project or in the middle of developing a project. I highly recommend opting for v5

**Reply by davebot9000** [reply]
Thank you for the prompt response Harmender, appreciate it!

**Reply by meideros** [reply]
I hope the web socket fonctionnality will be inside the v5 release !

**Reply by meideros** [reply]
I hope the web socket fonctionnality will be inside the v5 release !

**Reply by davebot9000** [reply]
It’s not going to be. This is according to what Virk said on Discord
